Contributing.md

```diff
@@ -173,7 +173,8 @@ Summary:
 - [Classes and Structs](#cpp-code-classes-and-structs)
 
 ## <a name="cpp-code-general"></a>General
-- The codebase currently uses C++17.
+- The codebase currently uses C++20, though not all compilers support all C++20 features.
+  - The minimum supported compilers are currently GCC 10, Clang 11, and MSVC 14.32.
```

Comment:
maybe just have a pointer to where the check is enforced in code/cmake/etc? we also currently state required VS version in readme and it's annoying to update / probably shouldn't .

Reply:
It's a bit less visual, and its only three numbers that we have to update. Although, I don't see a ton of reason in listing the required compilers in Contributing.md along with Readme.md.

Reply:
Ok, I've removed explicit version references in both readme and contributing.